pkg/storage/iterator.go

```diff
@@ -385,6 +385,7 @@ func fetchLazyChunks(ctx context.Context, chunks []*chunkenc.LazyChunk) error {
 	}
 	level.Debug(log).Log("msg", "loading lazy chunks", "chunks", totalChunks)
 
+	invalidChunks := make(chan *chunk.Fetcher)
```

Comment:
I am creating this channel so that later when we have to remove data for chunks which are invalid, we can iterate only few chunks rather than iterating all fetchers and chunks.